# HG changeset patch
# User Rik <octave@nomad.inbox5.com>
# Date 1339737064 25200
#      Thu Jun 14 22:11:04 2012 -0700
# Node ID 39a2e91a246e21afe182fe5709db6e5b928002ec
# Parent  10ed11922f1903a4c282e259f546f4956766695e
Fix failing logncdf tests by adding eps tolerance.

* logncdf.m: Add eps tolerance to tests.

diff --git a/scripts/statistics/distributions/logncdf.m b/scripts/statistics/distributions/logncdf.m
--- a/scripts/statistics/distributions/logncdf.m
+++ b/scripts/statistics/distributions/logncdf.m
@@ -68,26 +68,26 @@ function cdf = logncdf (x, mu = 0, sigma
     cdf(k) = stdnormal_cdf ((log (x(k)) - mu(k)) ./ sigma(k));
   endif
 
 endfunction
 
 
 %!shared x,y
 %! x = [-1 0 1 e Inf];
-%! y = [0, 0, 0.5, 1/2+1/2*erf((1/sqrt(2))*(1/sqrt(2))), 1];
-%!assert (logncdf (x, zeros (1,5), sqrt(2)*ones (1,5)), y)
-%!assert (logncdf (x, 0, sqrt(2)*ones (1,5)), y)
-%!assert (logncdf (x, zeros (1,5), sqrt(2)), y)
-%!assert (logncdf (x, [0 1 NaN 0 1], sqrt(2)), [0 0 NaN y(4:5)])
-%!assert (logncdf (x, 0, sqrt(2)*[0 NaN Inf 1 1]), [NaN NaN NaN y(4:5)])
-%!assert (logncdf ([x(1:3) NaN x(5)], 0, sqrt(2)), [y(1:3) NaN y(5)])
+%! y = [0, 0, 0.5, 1/2+1/2*erf(1/2), 1];
+%!assert (logncdf (x, zeros (1,5), sqrt(2)*ones (1,5)), y, eps)
+%!assert (logncdf (x, 0, sqrt(2)*ones (1,5)), y, eps)
+%!assert (logncdf (x, zeros (1,5), sqrt(2)), y, eps)
+%!assert (logncdf (x, [0 1 NaN 0 1], sqrt(2)), [0 0 NaN y(4:5)], eps)
+%!assert (logncdf (x, 0, sqrt(2)*[0 NaN Inf 1 1]), [NaN NaN NaN y(4:5)], eps)
+%!assert (logncdf ([x(1:3) NaN x(5)], 0, sqrt(2)), [y(1:3) NaN y(5)], eps)
 
 %% Test class of input preserved
-%!assert (logncdf ([x, NaN], 0, sqrt(2)), [y, NaN])
+%!assert (logncdf ([x, NaN], 0, sqrt(2)), [y, NaN], eps)
 %!assert (logncdf (single ([x, NaN]), 0, sqrt(2)), single ([y, NaN]), eps ("single"))
 %!assert (logncdf ([x, NaN], single (0), sqrt(2)), single ([y, NaN]), eps ("single"))
 %!assert (logncdf ([x, NaN], 0, single (sqrt(2))), single ([y, NaN]), eps ("single"))
 
 %% Test input validation
 %!error logncdf ()
 %!error logncdf (1,2)
 %!error logncdf (1,2,3,4)
